app/controllers/assignments_controller.rb

```diff
@@ -51,9 +53,9 @@ def set_assignment
     @assignment = Assignment.find(params[:id])
   end
 
-    def set_requirement
-      @requirement = Requirement.find(assignment_params[:requirement_id])
-    end
+  def set_requirement
+    @requirement = Requirement.find(params[:requirement_id])
```

Comment:
Why do you not want to use the assignment strong params ?

http://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters

Reply:
in some views there was no requirement key to build the strong_params from, leading to a server error.